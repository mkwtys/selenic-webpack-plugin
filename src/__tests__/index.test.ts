import assert from 'assert'
import extractComments from 'extract-comments'
import MemoryFS from 'memory-fs'
import path from 'path'
import TerserWebpackPlugin from 'terser-webpack-plugin'
import webpack from 'webpack'
import Plugin from '../'

describe('Plugin', () => {
  it('need apply method', () => {
    assert(typeof new Plugin().apply === 'function')
  })

  function fixtureTest(fixtureName: string) {
    it(`fixtures: ${fixtureName}`, done => {
      const compiler = webpack({
        mode: 'production',
        entry: {
          index: path.resolve(__dirname, `fixtures/${fixtureName}/src`)
        },
        output: {
          path: path.resolve(__dirname, `fixtures/${fixtureName}/dist`),
          filename: '[name].js'
        },
        plugins: [
          new TerserWebpackPlugin({
            terserOptions: {
              compress: {
                dead_code: true,
                unused: true,
                warnings: true
              },
              output: {
                comments: false
              },
              sourceMap: false
            }
          }),
          new Plugin()
        ],
        optimization: {
          minimize: false
        }
      })
      compiler.outputFileSystem = new MemoryFS()
      compiler.run((err, stats) => {
        const compiled = stats.compilation.assets['index.js']
          ? stats.compilation.assets['index.js'].source()
          : ''
        const comments = extractComments(compiled)
        const licenseHeader = comments.length ? `${comments[0].raw}` : ''
        expect(licenseHeader).toMatchSnapshot()
        done()
      })
    })
  }

  fixtureTest('minimal')
  fixtureTest('private')
  fixtureTest('deps')
})
