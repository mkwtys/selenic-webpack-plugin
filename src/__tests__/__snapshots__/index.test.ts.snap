// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugin fixtures minimal config: deps 1`] = `
"/**
 * @license
 *
 * deps:
 *   version: 0.0.0
 *   license: MIT
 *   author: a <a@example.com>
 *
 * dep:
 *   version: 1.0.0
 *   license: MIT
 *   author: a <a@example.com>
 *   contributors: c1 <c1@example.com>, c2 <c2@example.com>
 *
 */"
`;

exports[`Plugin fixtures minimal config: homepage 1`] = `
"/**
 * @license
 *
 * homepage:
 *   version: 1.0.0
 *   license: MIT
 *   author: a <a@example.com>
 *
 * github:
 *   version: 1.0.0
 *   license: MIT
 *   homepage: https://github.com/
 *
 * github-hash:
 *   version: 1.0.0
 *   license: MIT
 *   homepage: https://github.com/#hash
 *
 * github-hash-readme:
 *   version: 1.0.0
 *   license: MIT
 *   homepage: https://github.com/mkwtys/selenic-webpack-plugin
 *
 * hash:
 *   version: 1.0.0
 *   license: MIT
 *   homepage: https://example.com/#hash
 *
 * hash-readme:
 *   version: 1.0.0
 *   license: MIT
 *   homepage: https://example.com/#readme
 *
 */"
`;

exports[`Plugin fixtures minimal config: minimal 1`] = `
"/**
 * @license
 *
 * a:
 *   version: 0.0.0
 *   license: MIT
 *   author: a <a@example.com>
 *
 */"
`;

exports[`Plugin fixtures minimal config: multi-versions 1`] = `
"/**
 * @license
 *
 * multi-versions:
 *   version: 0.0.0
 *   license: MIT
 *   author: a <a@example.com>
 *
 * a:
 *   version: 1.0.0
 *   license: MIT
 *   author: a <a@example.com>
 *
 * all:
 *   version: 1.0.0
 *   license: MIT
 *   author: all <all@example.com>
 *
 * b:
 *   version: 1.0.0
 *   license: MIT
 *   author: b <b@example.com>
 *
 * c:
 *   version: 1.0.0
 *   license: MIT
 *   author: c <c@example.com>
 *
 * d:
 *   version: 1.0.0, 2.0.0
 *   license: MIT
 *   author: d <d@example.com>
 *
 *   version: 3.0.0
 *   license: ISC
 *   author: d <d@example.com>
 *
 */"
`;

exports[`Plugin fixtures minimal config: private 1`] = `""`;

exports[`Plugin fixtures splitChunks: deps 1`] = `
"/**
 * @license
 *
 * deps:
 *   version: 0.0.0
 *   license: MIT
 *   author: a <a@example.com>
 *
 */"
`;

exports[`Plugin fixtures splitChunks: deps 2`] = `
"/**
 * @license
 *
 * dep:
 *   version: 1.0.0
 *   license: MIT
 *   author: a <a@example.com>
 *   contributors: c1 <c1@example.com>, c2 <c2@example.com>
 *
 */"
`;

exports[`Plugin fixtures splitChunks: homepage 1`] = `
"/**
 * @license
 *
 * homepage:
 *   version: 1.0.0
 *   license: MIT
 *   author: a <a@example.com>
 *
 */"
`;

exports[`Plugin fixtures splitChunks: homepage 2`] = `
"/**
 * @license
 *
 * github:
 *   version: 1.0.0
 *   license: MIT
 *   homepage: https://github.com/
 *
 * github-hash:
 *   version: 1.0.0
 *   license: MIT
 *   homepage: https://github.com/#hash
 *
 * github-hash-readme:
 *   version: 1.0.0
 *   license: MIT
 *   homepage: https://github.com/mkwtys/selenic-webpack-plugin
 *
 * hash:
 *   version: 1.0.0
 *   license: MIT
 *   homepage: https://example.com/#hash
 *
 * hash-readme:
 *   version: 1.0.0
 *   license: MIT
 *   homepage: https://example.com/#readme
 *
 */"
`;

exports[`Plugin fixtures splitChunks: minimal 1`] = `
"/**
 * @license
 *
 * a:
 *   version: 0.0.0
 *   license: MIT
 *   author: a <a@example.com>
 *
 */"
`;

exports[`Plugin fixtures splitChunks: minimal 2`] = `""`;

exports[`Plugin fixtures splitChunks: multi-versions 1`] = `
"/**
 * @license
 *
 * multi-versions:
 *   version: 0.0.0
 *   license: MIT
 *   author: a <a@example.com>
 *
 */"
`;

exports[`Plugin fixtures splitChunks: multi-versions 2`] = `
"/**
 * @license
 *
 * a:
 *   version: 1.0.0
 *   license: MIT
 *   author: a <a@example.com>
 *
 * all:
 *   version: 1.0.0
 *   license: MIT
 *   author: all <all@example.com>
 *
 * b:
 *   version: 1.0.0
 *   license: MIT
 *   author: b <b@example.com>
 *
 * c:
 *   version: 1.0.0
 *   license: MIT
 *   author: c <c@example.com>
 *
 * d:
 *   version: 1.0.0, 2.0.0
 *   license: MIT
 *   author: d <d@example.com>
 *
 *   version: 3.0.0
 *   license: ISC
 *   author: d <d@example.com>
 *
 */"
`;

exports[`Plugin fixtures splitChunks: private 1`] = `""`;

exports[`Plugin fixtures splitChunks: private 2`] = `""`;
